### DISCUSSION TITLE
Accessing the record in the form function

### DISCUSSION THREAD

**Discussion by Wulfheart** [filamentphp/filament#6027] [Help] [answered]
Hello,

I would like to display different forms based on the state of the record. Therefore I would like to do an if else in the form method. However, the record is not yet defined in the static method. Is there any way to access it so I don’t have to have one big definition where I hide fields based on the state with a callback function?

Thank you in advance.

**Comment by danharrin** [comment, marked as answer]
You could hide fields conditionally. Or, you could skip the `form()` method on the Resource class and instead use `getFormSchema()` on the Edit page, which has access to `$this->record`.

In v3, the record is accessible in the resource :)

**Reply by Wulfheart** [reply]
![image](https://user-images.githubusercontent.com/25671390/227707946-eb74bea1-7523-49f3-a5e1-79dc5fe39f69.png)

Note: I know that it depends on Livewire v3.

**Reply by danharrin** [reply]
😅
